server/model/encryption.go

```diff
@@ -0,0 +1,27 @@
+package model
+
+// EncryptionBuilder is user API to obtain correctly configured encryption
+type EncryptionBuilder interface {
+	WithClient(client EncryptionClient) EncryptionBuilder
+	Build()
+}
+
+type EncryptionServiceBuilder interface {
+	WithClients(clients []EncryptionClient) EncryptionServiceBuilder
+	Build() EncryptionService
+}
+
+type EncryptionService interface {
+	Encrypt(plaintext string, associatedData string) string
```

Comment:
does associatedData change for each call? imho it should be just:

```go
Encrypt(plaintext string) string
Decrypt(ciphertext string) string
```

Reply:
yes, the common practise is using object id to encrypt its fields